src/core/weights.js

```diff
@@ -48,19 +57,72 @@ export function copy(w: Weights): Weights {
   };
 }
 
-/**
- * Merge multiple Weights together.
+/** Merge multiple Weights together.
+ *
+ * The resultant Weights will have every weight specified by each of the input
+ * weights.
+ *
+ * When there are overlaps (i.e. the same address is present in two or more of
+ * the Weights), then the appropriate resolver will be invoked to resolve the
+ * conflict. The semantics are similar to Array.reduce, where the accumulated
+ * weight and the current weight are both passed as arguments to the resolver.
+ * For the first conflict, the "accumulated weight" is equal to the first
+ * weight matching that address. The final accumulated weight will be included
+ * in the resultant Weights.
```

Comment:
As [mentioned earlier][1] (and also in personal communication before
this PR was sent), I think that left-to-right accumulator semantics are
confusing here. By saying that “the accumulated weight and the current
weight are both passed as arguments to the resolver”, the client must
consider two distinct concepts: a notion of _time_ (“current” weight)
and a notion of _state_ (“accumulated” weight). This is inessential.

It would be simpler to say:

> The reducer takes two weights and combines them to return a new
> weight.

Then, we don’t have to think about whether we’re folding left- or
right-associatively, and don’t have to think about any notion of
currency.

[1]: https://github.com/sourcecred/sourcecred/pull/1597#discussion_r372164566

Reply:
Done.